src/core/mac/mac.cpp

```diff
@@ -828,12 +887,17 @@ otError Mac::PrepareDataRequest(TxFrame &aFrame)
     otError  error = OT_ERROR_NONE;
     Address  src, dst;
     uint16_t fcf;
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+    Neighbor *parent = &Get<Mle::MleRouter>().GetParentCandidate();
```

Comment:
```suggestion
    Neighbor *parent = &Get<DataPollSender>().GetParent();
```

Reply:
`GetParent` is a private method in `DataPollSender` and cannot be accessed from `Mac`. Shall we add a friend class for `DataPollSender` or change it to a public method? What's the different between getting parent from `MleRouter` and `DataPollSender`?